### PR TITLE
LOOP-1492: Therapy Settings: Add Insulin model section

### PR DESCRIPTION
**Pull request by rickpasetto**
This adds the Insulin Model view to Therapy Settings.  
Notes: 
- I've moved a bunch of Insulin Model-related structs/extensions/etc into `LoopKit`.  This was necessary to support consistent display of this screen in `LoopKitUI` and in `Loop`
- I broke up the `InsulinModelSelection` screen into some reusable components, but stopped short of making a whole "InsulinModelSelection" reusable view. I figure it was a bit overkill, but if folks think it would be helpful I can go one step further and do this.

See also https://github.com/tidepool-org/LoopKit/pull/155

[LOOP-1492](https://tidepool.atlassian.net/browse/LOOP-1492)

### CODE REVIEW

File: Loop/Views/InsulinModelSelection.swift [hunk 1]
```diff
@@ -98,35 +94,35 @@ struct InsulinModelSelection: View, HorizontalSizeClassOverride {
                         )
                         .frame(height: 170)
 
-                        InsulinModelSettingsRow(
-                            title: Text("Rapid Acting - Adult", comment: "Title for rapid acting adult insulin model"),
-                            description: Text("This model assumes peak insulin activity at 75 minutes.", comment: "Description for rapid acting adult insulin model"),
+                        CheckmarkListItem(
+                            title: Text(InsulinModelSettings.exponentialPreset(.humalogNovologAdult).title),
+                            description: Text(InsulinModelSettings.exponentialPreset(.humalogNovologAdult).subtitle),
```
Comment (mpangburn) [inline Loop/Views/InsulinModelSelection.swift:99]: Your LoopKit PR seems to extract the old titles/descriptions (which I failed to remove when redoing this screen); the ones included in InsulinModelSelection here are the updated versions per the design spec

Comment (novalegra) [reply Loop/Views/InsulinModelSelection.swift:99]: +1

Comment (novalegra) [reply Loop/Views/InsulinModelSelection.swift:99]: +1

Comment (rickpasetto) [reply Loop/Views/InsulinModelSelection.swift:99]: Will fix and update. (and good catch!!)

**Review by novalegra — APPROVED** [review]
LGTM

**Pull request merged**